source/Halibut.Tests/TestServices/Async/IAsyncEchoService.cs

```diff
@@ -0,0 +1,15 @@
+using System.Threading.Tasks;
+
+namespace Halibut.Tests.TestServices.Async
+{
+    public interface IAsyncEchoService
```

Comment:
Do we now end up with 4 different interfaces to support async and client cancellation if we add new interfaces rather than reusing existing e.g. `IEchoService`, `IClientEchoService`, `IAsyncEchoService` and `IAsyncClientEchoService`

If we just added async overloads to `IClientEchoService` we don't add all these variations.

Reply:
Renamed this to IAsynceClientEchoService